File: dao/src/main/java/com/alta_v2/game/dao/provider/ConnectionSourceProvider.java
package com.alta_v2.game.dao.provider;

import com.google.inject.Provider;
import com.j256.ormlite.jdbc.JdbcConnectionSource;
import com.j256.ormlite.support.ConnectionSource;
import lombok.extern.log4j.Log4j2;

import java.sql.SQLException;

@Log4j2
public class ConnectionSourceProvider implements Provider<ConnectionSource> {

    @Override
    public ConnectionSource get() {
        try {
            return new JdbcConnectionSource("jdbc:sqlite:data.db3");
        } catch (SQLException e) {
            log.error(e.getMessage());
            return null;
        }
    }
}

File: dao/src/main/java/com/alta_v2/game/dao/provider/ConnectionSourceProvider.kt
package com.alta_v2.game.dao.provider

import com.alta_v2.DB_PATH
import com.google.inject.Provider
import com.j256.ormlite.jdbc.JdbcConnectionSource
import com.j256.ormlite.support.ConnectionSource
import mu.KotlinLogging
import java.sql.SQLException

class ConnectionSourceProvider : Provider<ConnectionSource?> {

    private val log = KotlinLogging.logger {  }

    override fun get(): ConnectionSource? {
        return try {
            val dbPath = System.getProperty(DB_PATH)

            return if (dbPath != null) {
                JdbcConnectionSource("jdbc:sqlite:$dbPath")
            } else {
                JdbcConnectionSource("jdbc:sqlite:data.db3")
            }
        } catch (e: SQLException) {
            log.error("Failed to create jdbc connection source", e)
            null
        }
    }
}